### PR TITLE
Achataigner/fix cmake for ubuntu 20 04

### PR DESCRIPTION
**Pull request by achataigner**

### CODE REVIEW

**Comment by hansenms** [comment]
We don't have cmake 3.23 in the environment file, so this will fail to build. Please check that the proposed changes build (just use the devcontainer or try to run the `build_images.sh` script.

File: CMakeLists.txt [hunk 1]
```diff
@@ -214,6 +212,8 @@ if (NOT DEFINED Boost_VERSION_STRING)
     set(Boost_VERSION_STRING "${Boost_MAJOR_VERSION}.${Boost_MINOR_VERSION}.${Boost_SUBMINOR_VERSION}")
 endif()
 
+# date library: https://github.com/HowardHinnant/date/
+find_package(date REQUIRED)
```
Comment (hansenms) [inline CMakeLists.txt:216]: Why is this needed? Is this just to get an error if it is not installed?

Comment (achataigner) [reply CMakeLists.txt:216]: Yes it is just to get the error at cmake configuration rather than compilation.

**Comment by achataigner** [comment]
> We don't have cmake 3.23 in the environment file, so this will fail to build. Please check that the proposed changes build (just use the devcontainer or try to run the `build_images.sh` script.

On origin/master devcontainer does not work on my machine, here is the log:
[remoteContainers-2022-11-16T11-47-02.959Z.log](https://github.com/gadgetron/gadgetron/files/10022067/remoteContainers-2022-11-16T11-47-02.959Z.log)

Could you help me ?

Thanks

**Comment by hansenms** [comment]
Looks like your Docker mount points are not found on your host system. Says `source path does not exist: /var/run/docker.sock.`. Not sure what host system you are on, but that seems to be causing the problem.

**Comment by achataigner** [comment]
> ould disable the docker-from-docker feature in the devcontainer.json file. You will not be able to run Docker commands inside the devcontainer, but still may get you to debug the things you need.

I'm still on the origin/master branch.
I removed docker-from-docker and devcontainer configuration succeeds.
Now how can I test the build without docker-from-docker ?

Sorry for my lack of experience with docker.

**Comment by hansenms** [comment]
@achataigner once you are in the devcontainer, you can just run cmake (wither command line or CTRL+SHIFT+P and the type cmake to see configure, build, install commands). If you want to check that you can build docker images, you will have to do it outside that devcontainer environment (just run the build-images.sh script on a machine with Docker).

Attend the Friday meetings if you need a hand.

**Comment by achataigner** [comment]
I have still some issues, I think I've spent too much time on this, you can close this PR if you want.
Thanks for your help
